<!DOCTYPE html>
<html lang="en">

  <head>
    <meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

<meta property="og:title" content="Cropping Rasters" />
<meta property="og:description" content="It is often the case that the raster we are working with is not the exact size of the area from which our data are collected. It is a much easier situation if the raster is larger than the area than if you need to stitch together two raster Tiles to get all your data onto one extent. In my doctoral thesis work, the area of the southern Ozark mountains that my sites were in was not only straddling a boundary between existing rasters, it was also at the boundary of two UTM zones!" />
<meta property="og:type" content="article" />
<meta property="og:url" content="http://dyerlab/2016/03/cropping-rasters/" />


<meta property="og:updated_time" content="2016-03-18T00:00:00&#43;00:00"/>











<meta itemprop="name" content="Cropping Rasters">
<meta itemprop="description" content="It is often the case that the raster we are working with is not the exact size of the area from which our data are collected. It is a much easier situation if the raster is larger than the area than if you need to stitch together two raster Tiles to get all your data onto one extent. In my doctoral thesis work, the area of the southern Ozark mountains that my sites were in was not only straddling a boundary between existing rasters, it was also at the boundary of two UTM zones!">


<meta itemprop="dateModified" content="2016-03-18T00:00:00&#43;00:00" />
<meta itemprop="wordCount" content="468">



<meta itemprop="keywords" content="2gener,aflp,applied-population-genetics,araptus-attenuata,baja-california,blog,ces,cgd,code,default-calendar,documents,dogwood,dyerlab,edna,fitness,gdistance,gene-flow,genetic-structure,genomics,ggplot2,google,google-docs,graduate,gsl,gstudio,laboratory,landscape-genetics,lophocereus-schottii,main-menu,manuscripts,maps,markers,null-alleles,ole2015,online,online-publishing,paternity,people,phylogeography,polygon,popgraph,population-genetics,population-graphs,presentations,projects,protocols,publications,r,raster,reproducible-research,research,rstudio,servers,slack,software,spatial,stamova,sturgeon,sustainability,teaching,theory,tutorial,tutorials,uncategorized,unix,urbanization,vcu,vcugenetics2015,xcode," />



  <meta name="twitter:card" content="summary"/>



<meta name="twitter:title" content="Cropping Rasters"/>
<meta name="twitter:description" content="It is often the case that the raster we are working with is not the exact size of the area from which our data are collected. It is a much easier situation if the raster is larger than the area than if you need to stitch together two raster Tiles to get all your data onto one extent. In my doctoral thesis work, the area of the southern Ozark mountains that my sites were in was not only straddling a boundary between existing rasters, it was also at the boundary of two UTM zones!"/>
<meta name="twitter:site" content="@dyerlab"/>









<meta name="generator" content="Hugo 0.18.1" />


    <link rel="canonical" href="http://dyerlab/2016/03/cropping-rasters/">

    <title>Cropping Rasters | The Dyer Laboratory</title>

    <!-- combined, minified CSS -->
    <link href="http://dyerlab/css/style.css" rel="stylesheet" integrity="sha384-qRVpIj9hSzsBhmO8Y7YEKF2UFra2sJQtl9V/uFKKDvy&#43;Wjh9zgTku6VRgT8YdPoD" crossorigin="anonymous">

    

    

    

    
<link rel="stylesheet" href="/css/dyerlab.css">
<script>
  (function() {
    var cx = '005624429758912894427:ewy8fmekoli';
    var gcse = document.createElement('script');
    gcse.type = 'text/javascript';
    gcse.async = true;
    gcse.src = 'https://cse.google.com/cse.js?cx=' + cx;
    var s = document.getElementsByTagName('script')[0];
    s.parentNode.insertBefore(gcse, s);
  })();
</script>




  </head>

  <body>

    <div class="blog-masthead">
      <div class="container">
        <nav class="nav blog-nav">
          <a class="nav-link " href="http://dyerlab/">Home</a>
          
          
          <a class="nav-link" href="http://dyerlab/people/">People</a>
          
          <a class="nav-link" href="http://dyerlab/publications/">Publications</a>
          
          <a class="nav-link" href="http://dyerlab/texts/">Texts</a>
          
          
        </nav>
      </div>
    </div>

    <header class="blog-header">
      <div class="container">
        <h1 class="blog-title"><a href="http://dyerlab/" rel="home">The Dyer Laboratory</a></h1>
        <p class="lead blog-description">Illuminating the mechanics of genetic connectivity.</p>
      </div>
    </header>

    <div class="container">
      <div class="row">
        <div class="col-sm-8 blog-main">

          


<article class="blog-post">
  <header>
    <h2 class="blog-post-title"><a href="http://dyerlab/2016/03/cropping-rasters/">Cropping Rasters</a></h2>
    <p class="blog-post-meta"><time datetime="2016-03-18T00:00:00Z">Fri Mar 18, 2016</time> by Rodney J. Dyer in 
<i class="fa fa-folder" aria-hidden="true"></i>&nbsp;<a href="/categories/applied-population-genetics" rel="category tag">Applied Population Genetics</a>, <a href="/categories/code" rel="category tag">Code</a>, <a href="/categories/r" rel="category tag">R</a>


</p>
  </header>


<p>It is often the case that the raster we are working with is not the exact size of the area from which our data are collected. It is a much easier situation if the raster is larger than the area than if you need to stitch together two raster Tiles to get all your data onto one extent. In my doctoral thesis work, the area of the southern Ozark mountains that my sites were in was not only straddling a boundary between existing rasters, it was also at the boundary of two UTM zones! What a pain. Spatial data often consists of very large datasets. One way to minimize the amount of computational resources you are going to use in R is to select only the spatial regions (extents) that you are working in and get rid of the remaining data. This is particularly important if you are going to be doing isolation modeling for genetic connectivity as the cost surface raster needs to be translated into a connectivity network. Data outside your area of interest are unnecessarily taking both resources and time away from your work. To crop a raster, we need to identify the region we wish to keep as an extent object. An extent is a vector of length 4 defined as <code>c(xmin, xmax, ymin, ymax)</code> describing the boundaries of are of interest. Here is an extent for the arapat data set.require(gstudio)</p>
<pre class="r"><code>library(raster)
library(gstudio)
data(arapat)
lon.min &lt;- min(arapat$Longitude)
lon.max &lt;- max(arapat$Longitude)
lat.min &lt;- min(arapat$Latitude)
lat.max &lt;- max(arapat$Latitude)
e &lt;- extent( lon.min, lon.max, lat.min, lat.max )
e</code></pre>
<pre><code>## class       : Extent 
## xmin        : -114.2935 
## xmax        : -109.1263 
## ymin        : 23.0757 
## ymax        : 29.32541</code></pre>
<p>This is the exact boundaries of the data set. However, it is probably a good idea to not have your map cropped exactly to your boundaries but have your most extreme locations plotted within the map boundaries. So Iâ€™ll take an approximation (rounding up and down as necessary) to use.</p>
<pre class="r"><code>e &lt;- extent( -115, -108, 23, 30 )
e</code></pre>
<pre><code>## class       : Extent 
## xmin        : -115 
## xmax        : -108 
## ymin        : 23 
## ymax        : 30</code></pre>
<p>This can now be used as the area from within the alt raster that we want to keep.</p>
<pre class="r"><code>alt &lt;- raster( &quot;../../data/alt_22.tif&quot; )
bc &lt;- crop( alt, e )
bc</code></pre>
<pre><code>## class       : RasterLayer 
## dimensions  : 840, 840, 705600  (nrow, ncol, ncell)
## resolution  : 0.008333333, 0.008333333  (x, y)
## extent      : -115, -108, 23, 30  (xmin, xmax, ymin, ymax)
## coord. ref. : +proj=longlat +datum=WGS84 +no_defs +ellps=WGS84 +towgs84=0,0,0 
## data source : in memory
## names       : alt_22 
## values      : -202, 2774  (min, max)</code></pre>
<p>And if we plot it, we see that it is more properly scaled for the region we are interested in working on.</p>
<pre class="r"><code>plot(bc)</code></pre>
<p><img src="figures/unnamed-chunk-4-1.png" width="672" /></p>




  

  
  <hr>
  <footer>

  
    <section>
    <h4>Share</h4>
    <nav class="nav sharing-icons">
      <a class="nav-item" href="https://www.facebook.com/sharer/sharer.php?u=http%3a%2f%2fdyerlab%2f2016%2f03%2fcropping-rasters%2f" title="Share on Facebook"><span class="fa fa-facebook fa-2x" aria-hidden="true"></span></a>
      <a class="nav-item" href="https://plus.google.com/share?url=http%3a%2f%2fdyerlab%2f2016%2f03%2fcropping-rasters%2f" title="Share on Google+"><span class="fa fa-google-plus fa-2x" aria-hidden="true"></span></a>
      
      <a class="nav-item" href="https://twitter.com/home?status=http%3a%2f%2fdyerlab%2f2016%2f03%2fcropping-rasters%2f" title="Tweet this"><span class="fa fa-twitter fa-2x"></span></a>
    </nav>
  </section>

  

  
  </footer>
  

</article> 



        </div> <!-- /.blog-main -->

        <aside class="col-sm-3 offset-sm-1 blog-sidebar">
  
  
  

  <section class="sidebar-module  sidebar-module-inset">
    <h4>Recent Posts</h4>
    <ol class="list-unstyled">
      
      <li><a href="/2017/01/new-server-new-site-first-post/">New Server, New Site, First Post!</a></li>
      
      <li><a href="/2017/01/new-position/">New Position</a></li>
      
      <li><a href="/2016/12/life-sciences-graduation/">Life Sciences Graduation</a></li>
      
      <li><a href="/2016/11/talk-at-temple/">Talk at Temple!</a></li>
      
      <li><a href="/2016/10/spatial-data-sources/">Spatial Data Sources</a></li>
      
    </ol>
  </section>

  
  
  <section class="sidebar-module sidebar-module-inset">
  <h5>Search Dyerlab</h5>
  <gcse:searchbox-only></gcse:searchbox-only>
  </section>
  
  
  <section class="sidebar-module sidebar-module-inset">
    <h4>Links</h4>
    <ol class="list-unstyled">
      
      <li><a href="http://dyerlab.com/categories">Blog Tags</a></li>
      
      <li><a href="http://dyerlab.com/rstudio">RStudio</a></li>
      
      <li><a href="https://github.com/dyerlab">Github</a></li>
      
    </ol>
  </section>
  
  
  <div class="twitter-timeline">
    <a class="twitter-timeline" href="https://twitter.com/Dyerlab" data-height="1200">Tweets by Dyerlab</a><script async src="//platform.twitter.com/widgets.js" charset="utf-8"></script>
  </div>
</aside>


      </div> <!-- /.row -->
    </div> <!-- /.container -->

    <footer class="blog-footer">
      <p>
      
      &copy; by 2017 <a href="mailto:rjdyer@vcu.edu">R.J. Dyer</a>.
      
      </p>
      <p>
      <a href="#">Back to top</a>
      </p>
    </footer>

  </body>

</html>
